app/src/main/java/io/appium/uiautomator2/utils/AlertHelpers.java

```diff
@@ -16,30 +16,30 @@
 
 package io.appium.uiautomator2.utils;
 
-import android.app.Dialog;
 import android.support.annotation.Nullable;
 import android.support.test.uiautomator.By;
 import android.support.test.uiautomator.UiObject2;
 import android.widget.Button;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
-import java.util.regex.Pattern;
 
 import io.appium.uiautomator2.common.exceptions.InvalidElementStateException;
 import io.appium.uiautomator2.common.exceptions.NoAlertOpenException;
 
 import static android.text.TextUtils.join;
 import static io.appium.uiautomator2.utils.Device.getUiDevice;
+import static org.apache.commons.lang.StringUtils.isBlank;
 
 public class AlertHelpers {
-    private static UiObject2 getAlert() {
-        final List<UiObject2> dialogs = getUiDevice().findObjects(By.clazz(Dialog.class));
-        if (dialogs.isEmpty()) {
-            throw new NoAlertOpenException();
-        }
-        return dialogs.get(0);
+    private static final String alertButtonResIdPattern = "^android:id/button\\d+$";
+    private static final String alertContentResId = "android:id/content";
```

Comment:
So should we consider any view with id ```context``` as a dialog?
![image](https://user-images.githubusercontent.com/3530074/39543518-1b170c80-4e54-11e8-944b-484caa2f9ab2.png)